### PR TITLE
Add Dog muchies

### PR DESCRIPTION
**Pull request by tomsons26**

### CODE REVIEW

File: rules/allied-infantry.yaml [hunk 1]
```diff
@@ -144,6 +144,40 @@ snipe:
...
+		Queue: Infantry
+		BuildAtProductionType: Soldier
+		BuildPaletteOrder: 90
+		Prerequisites: ~gatech, ~gapile
+	Valued:
+		Cost: 500
+	DisguiseToolTip:
+		Name: Spy
+		GenericName: Soldier
+		Description: Infiltrates enemy structures for intel or\nsabotage. Exact effect depends on the\nbuilding infiltrated.\n  Strong vs Nothing\n  Weak vs Everything\n  Special Ability: Disguised
```
Comment (Phrohdoh) [inline rules/allied-infantry.yaml]: Can you change `Disguised` to `Disguises`, please?

File: rules/defaults.yaml [hunk 2]
```diff
@@ -5,7 +5,7 @@
 	Selectable:
 		Priority: 3
 	TargetableBuilding:
-		TargetTypes: Ground, C4, DetonateAttack
+		TargetTypes: Ground, C4, DetonateAttack, Structure, SpyInfiltrate
```
Comment (Phrohdoh) [inline rules/defaults.yaml]: Why is `Structure` being added here?

**Comment by Phrohdoh** [comment]
These changes will now need to be made against the file structure.

File: ra2/rules/allied-infantry.yaml [hunk 3]
```diff
@@ -144,6 +144,40 @@ snipe:
 		Weapon: awp
 	AttackFrontal:
 
+spy:
+	Inherits: ^Infantry
+	Buildable:
+		Queue: Infantry
+		BuildAtProductionType: Infantry
+		BuildPaletteOrder: 90
+		Prerequisites: ~gatech, ~infantry.allies
```
Comment (Phrohdoh) [inline ra2/rules/allied-infantry.yaml]: This needs to use the correct faction.

**Comment by Iamgoofball** [comment]
>dog munchies

10/10

**Pull request merged**